cmd/testvectors/main.go

```diff
@@ -0,0 +1,604 @@
+	spendAgg := sdkMultisig.Spend(0, privkeys[0], principalMultisig, destination, Amount, 0, opts...)
+
+	// add an individual test vector for each signing operation
+	// one list per tx type so we can assemble the final list in order
+	// start with the first operation
+	// three m-length lists plus one additional, final, aggregated self-spawn tx
+	txListSpawn := make([]TxPair, m)
+	txListSelfSpawn := make([]TxPair, m)
+	txListSpend := make([]TxPair, m)
+
+	// multisig txs are valid as standalone only if idx==m-1, i.e., it's the final part
+	txListSpawn[0] = TxPair{txtype: Spawn, tx: spawnAgg.Raw(), valid: m == 1, part: 0}
+	txListSelfSpawn[0] = TxPair{txtype: SelfSpawn, tx: selfSpawnAgg.Raw(), valid: m == 1, part: 0}
+	txListSpend[0] = TxPair{txtype: Spend, tx: spendAgg.Raw(), valid: m == 1, part: 0}
+
+	// now add a test vector for each additional required signature
+	// note: this assumes signer n has the signed n-1 tx
+	for signerIdx := uint8(1); signerIdx < m; signerIdx++ {
+		spawnAgg.Add(*sdkMultisig.Spawn(signerIdx, privkeys[signerIdx], principalMultisig, templateAddress, spawnArgsMultisig, 0, opts...).Part(signerIdx))
+		selfSpawnAgg.Add(*sdkMultisig.SelfSpawn(signerIdx, privkeys[signerIdx], templateAddress, m, pubkeysEd[:n], 0, opts...).Part(signerIdx))
+		spendAgg.Add(*sdkMultisig.Spend(signerIdx, privkeys[signerIdx], principalMultisig, destination, Amount, 0, opts...).Part(signerIdx))
+
+		// only the final, fully aggregated tx is valid
+		txListSpawn[signerIdx] = TxPair{txtype: Spawn, tx: spawnAgg.Raw(), valid: signerIdx == m-1, part: signerIdx}
+		txListSelfSpawn[signerIdx] = TxPair{txtype: SelfSpawn, tx: selfSpawnAgg.Raw(), valid: signerIdx == m-1, part: signerIdx}
+		txListSpend[signerIdx] = TxPair{txtype: Spend, tx: spendAgg.Raw(), valid: signerIdx == m-1, part: signerIdx}
+	}
+
+	// assemble the final list of txs in order: spawn, self-spawn, final aggregated self-spawn to apply, spend
+	txList = append(txList, txListSpawn...)
+	txList = append(txList, txListSelfSpawn...)
+	txList = append(txList, TxPair{txtype: Ignore, tx: selfSpawnAgg.Raw()})
+	txList = append(txList, txListSpend...)
+
+	return txList
+}
+
+const Amount = uint64(constants.OneSmesh)
+
+func generateKeys(n int) ([]signing.PublicKey, []core.PublicKey, []ed25519.PublicKey, []ed25519.PrivateKey) {
+	// generate the required set of keypairs
+
+	// frustratingly, we need the same list of pubkeys in multiple formats
+	// https://github.com/spacemeshos/go-spacemesh/issues/6061
+	pubkeysSigning := make([]signing.PublicKey, n)
+	pubkeysCore := make([]core.PublicKey, n)
+	pubkeysEd := make([]ed25519.PublicKey, n)
+	privkeys := make([]signing.PrivateKey, n)
+	for i := 0; i < n; i++ {
+		pubkeysEd[i], privkeys[i] = getKey()
+		pubkeysCore[i] = types.BytesToHash(pubkeysEd[i])
+		pubkeysSigning[i] = signing.PublicKey{PublicKey: pubkeysEd[i]}
+	}
+	return pubkeysSigning, pubkeysCore, pubkeysEd, privkeys
+}
```

Comment:
See my other comment. I feel this could be simpler. `getKey` isn't needed at all if `signing.EdSigner()` is used instead:

```suggestion
func generateKeys(n int) []*signing.EdSigner {
	identities := make([]*signing.EdSigner, 0, n)
	for range n {
		identity, err := signing.NewEdSigner()
		log.With().Fatal("failed creating identity", log.Err(err))
		identities = append(identities, identity)
	}
	return identities
}
```